src/main/kotlin/me/zeroeightsix/kami/feature/command/NbtCommand.kt

```diff
@@ -0,0 +1,221 @@
+    widgetsTrees,
+    contentRegion,
+    childWindows,
+    windowsUtilities,
+    itemWidgetsUtilities,
+    columns,
+    cursorLayout
+// this comment is here to prevent intelli'ntJ from formatting this against the kotlin conventions
+{
+    override fun render(matrices: MatrixStack?, mouseX: Int, mouseY: Int, delta: Float) {
+        super.render(matrices, mouseX, mouseY, delta)
+
+        KamiHud.frame(matrices!!) {
+            this()
+        }
+    }
+
+    operator fun invoke() {
+        dsl.window("NBT") {
+            if (button("Copy", Vec2(windowContentRegionWidth, 0)))
+                copyTagToClipboard(tag)
+            beginChild("##nbtPane", border = true)
+            columns(2)
+            showTree("NBT", tag)
+            columns(1)
+            endChild()
+        }
+    }
+
+    private fun showTree(tagName: String, tag: Tag, appendix: Int = 0) {
+        alignTextToFramePadding()
+        var curApp = appendix
+        if (treeNodeEx("$tagName##$curApp", TreeNodeFlag.NoTreePushOnOpen.i)) {
+            nextColumn()
+            nextColumn()
+            treePush()
+            when (tag) {
+                is CompoundTag -> {
+                    for (key in tag.keys)
+                        tag[key]?.let {
+                            if (tagIsDeep(it))
+                                showTree(key, it, ++curApp)
+                            else
+                                dsl.withStyleColor(Col.Text, tagColor(it)) {
+                                    nbtLabelText(key, it.asString())
+                                }
+                        }
+                }
+                is AbstractListTag<*> -> {
+                    for (t in tag)
+                        if (tagIsDeep(t))
+                            showTree("[array entry]", t, ++curApp)
+                        else
+                            dsl.withStyleColor(Col.Text, tagColor(t)) {
+                                nbtLabelText("[array entry]", t.asString())
+                            }
+                }
+                else -> {
+                    textColored(tagColor(tag), tag.asString())
+                    separator()
+                }
+            }
+            treePop()
+        }
+        nextColumn()
+        nextColumn()
+    }
+
+    private fun tagIsDeep(tag: Tag): Boolean {
+        val test = { t: Tag -> t is CompoundTag || t is ListTag }
+        return when (tag) {
+            is CompoundTag -> tag.keys.size > 1 || tag.keys.map { tag[it]!! }.any(test)
+            is AbstractListTag<*> -> tag.size > 1 || tag.any(test)
+            else -> false
+        }
+    }
+
+    private fun copyTagToClipboard(tag: Tag) {
+        mc.keyboard.clipboard = tag.asString()
+    }
+
+    private fun tagColor(tag: Tag) = when (tag) {
+        is AbstractNumberTag -> Vec4(1f, 1f, 0f, 1f)
+        is StringTag -> Vec4(0f, 1f, 0f, 1f)
+        else -> defaultColor
+    }
+
+    private fun nbtLabelText(k: String, v: String) {
```

Comment:
```suggestion
    private fun nbtLabelText(k: String, v: String): Unit = with(ImGui) {
```
(The `:Unit` is because kotlin type checking will otherwise fail in the recursive call)